### PR TITLE
Responses to teams are not correctly updated when a team is renamed #3404

### LINKED ISSUE #3404: Responses to teams are not correctly updated when a team is renamed
When updating teams and sections in the enrollment page, and the entire team is moved to a different section, FeedbackResponseLogic fails to correctly update the response's section where the team is the recipient.

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #3404

### CODE REVIEW

**Comment by kanghj** [comment]
@LowWeiLin

File: src/main/java/teammates/logic/automated/FeedbackSubmissionAdjustmentAction.java [hunk 1]
```diff
@@ -94,4 +104,171 @@ public boolean execute() {
...
+    /**
+     * Using the enrollment list, return the list of teams which are renamed.
+     * Updates the set of students who are modified by the enrollment.
+     * Updates the mapping from old section and team names to new section and team names.
+     * @param enrolmentList
+     * @param modifiedStudents
+     * @param oldSectionTeamToNewSectionTeamMap
+     * @return
+     */
+    private Set<String> extractDataFromEnrollmentList(
```
Comment (LowWeiLin) [inline src/main/java/teammates/logic/automated/FeedbackSubmissionAdjustmentAction.java]: Rename accordingly to what the function does..

**Comment by LowWeiLin** [comment]
Need to consider what happens to comments to teams when the team name changes.

**Comment by LowWeiLin** [comment]
Other things to consider when team name changes:
MCQ/MSQ has the option to generate options from team names
Permissions related to section/teams

**Pull request closed without merging**